osf/models/spam.py

```diff
@@ -266,3 +280,33 @@ def do_check_spam(self, author, author_email, content, request_headers, update=T
                 self.spam_data['who_flagged'] = 'oopspam'
                 self.spam_data['oopspam_data'] = oopspam_details
         return akismet_is_spam or oopspam_is_spam
+
+    def moderate_domains(self, content, confirm_spam=False):
+        domains = re.findall(
+            settings.DOMAIN_REGEX,
+            content
+        )
+        if not domains:
+            return False
+
+        from osf.models import NotableDomain
+        if confirm_spam and NotableDomain.has_spam_domain(content):
+            self.confirm_spam(save=True)
+        else:
```

Comment:
why is this an `else`? We can't both spam something based on domains *and* add new references?

Reply:
Good point, I will change it so it will capture new all domains and mark them as unknown even if only a single domain is spam.